magma/wire_container.py

```diff
@@ -122,9 +122,9 @@ def trace(self, skip_self=True):
             return self._driver.trace(skip_self=False)
         if not skip_self and (self._bit.is_output() or self._bit.is_inout()):
             return self._bit
-        if not skip_self and self._bit.has_children():
-            # Could be an Array driven non-bulk, so check if children can be
-            # traced
+        if (not skip_self and self._bit.has_children() and
+                self._bit.driven()):
+            # Driven non-bulk, so trace children
```

Comment:
nit: I've been preferring this style lately (it's also the style our clients use)

```suggestion
        if (
            not skip_self and self._bit.has_children() and self._bit.driven()
        ):
            # Driven non-bulk, so trace children
```